raddb/sites-available/ldap_sync

```diff
@@ -27,194 +102,391 @@
+			#  string data, so will want mapping to an integer attribute
+			#  for processing.
+			#
+			#  Firstly, one based on a naming context which covers all
+			#  user objects.  This will return results when objects are
+			#  added, modified or restored from the Deleted Objects
+			#  container.
+			#
+#			sync {
+#				base_dn = 'cn=Users,dc=example,dc=com'
+#				filter = '(userAccountControl:1.2.840.113556.1.4.803:=512)'
+#				scope = 'sub'
+#				attrs = 'userAccountControl'
+#				attrs = 'sAMAccountName'
+#
+#				update {
+#					&User-Name := 'sAMAccountName'
+#					&Tmp-Integer-0 := 'userAccountControl'
+#				}
+#			}
 
-			#  Specify the list of attributes to return in entries,
-			#  one config item per LDAP attribute.
 			#
-			#  If no attr config items are specified, then all attributes
-			#  will be returned.
-#			attr = 'cn'
-#			attr = 'foo'
+			#  Secondly, if you have the Recycle Bin enabled in Active
+			#  Directory and wish to be notified about deleted objects,
+			#  add a search covering the Deleted Objects container.
+			#
+			#  This will return results when an object is deleted, however
+			#  the DN and CN of the object are changed.  The attribute
+			#  lastKnownParent identifies where the object was deleted
+			#  from.  However, lastKnownParent may not be returned when
+			#  searching the Global Catalog.
+			#
+			#  Ensure this sync also includes the attribute isDeleted which
+			#  is used to direct processing of packets received to the
+			#  recv Delete section.
+			#
+#			sync {
+#				base_dn = "CN=Deleted Objects,dc=example,dc=com"
+#				filter = '(userAccountControl:1.2.840.113556.1.4.803:=512)'
+#				scope = "one"
+#				attrs = 'isDeleted'
+#				attrs = 'sAMAccountName'
+#				attrs = 'userAccountControl'
+#				attrs = 'lastKnownParent'
+#
+#				update {
+#					&User-Name := 'sAMAccountName'
+#					&Tmp-Integer-0 := 'userAccountControl'
+#					&Tmp-String-0 := 'lastKnownParent'
+#				}
+#			}
 
-			update {
-				&User-Name := 'cn'
-				&Password.With-Header := 'userPassword'
+			global {
+				ldap_debug = 0x00
 			}
 		}
-
-		sync {
-			base_dn = "ou=groups,dc=example,dc=org"
-			filter = "(objectClass=unixGroup}"
-		}
 	}
 
+	#
 	#  Provides FreeRADIUS with the last cookie value we received for the sync
 	#
+	#  This only applies to directories implementing RFC4533, such as OpenLDAP with
+	#  the syncprov overlay enabled.
+	#
+	#  For other directories, this will be called prior to the search query being
+	#  sent to the server, so could be used for any initial setup of cache datastores.
+	#
 	#  A request will be generated with the following attributes:
 	#
-	#  - &request.LDAP-Sync-DN		the base_dn of the sync.
-	#  - &request.LDAP-Sync-Filter		the filter of the sync (optional).
-	#  - &request.LDAP-Sync-Scope		the scope of the sync (optional).
-	#  - &request.LDAP-Sync-attr		the attributes returned by the sync (optional).
+	#  - &request.LDAP-Sync.DN		the base_dn of the sync.
```

Comment:
As these attributes creates in a TLV, or was this a search and replace when moving the LDAP attributes into a protocol specific dictionary?

Reply:
Looks like a TLV.  That's fine.